alacritty/src/main.rs

```diff
@@ -1,6 +1,8 @@
 //! Alacritty - The GPU Enhanced Terminal.
 
+#![warn(rust_2018_idioms, future_incompatible)]
```

Comment:
Why warn and not deny?

Reply:
Well, they are warnings by their nature, so I decided to nor abort on them for normal compilations, since if future rust will add more stuff into those checks and they trigger on more code, it'll be impossible to build alacritty, thus breaking packaging, since new rust won't be able to build old alacritty, since rust fine tuned warnings. However since we error on warnings when running clippy, we will error on them as well, and in the end there's no way we can miss anything from those warnings on CI.

Reply:
> However since we error on warnings when running clippy, we will error on them as well

Those are rust warnings though, not clippy warnings? Have you confirmed this is actually the case? After all it wasn't erroring on the other warnings you've set to deny in this PR either. Since `#[deny(warnings)]` does not disable all warnings but instead just disables everything that is in the "warnings" group by default.

But I'm perfectly fine with putting that behind a cfg and keeping it as only a warning for normal compilation. That way people will see it and at the same time it will fail CI.

Reply:
>Those are rust warnings though, not clippy warnings? Have you confirmed this is actually the case? After all it wasn't erroring on the other warnings you've set to deny in this PR either. Since #[deny(warnings)] does not disable all warnings but instead just disables everything that is in the "warnings" group by default.

Yes, but we start error on rustc warnings when running clippy on that PR, so everything will be aborted as you want with `deny`.

Reply:
Hm, and I think keeping those as warnings for normal builds instead of just adding them to `deny` when running clippy is better, since it's easier to notice them.

Reply:
> Yes, but we start error on rustc warnings when running clippy on that PR, so everything will be aborted as you want with deny.

We don't error on rustc warnings, we error on rustc lints that are usually warnings. Your current version will just spit out the warnings in clippy mode and keep going.

Reply:
Should be good, I've just confirmed myself that `deny(warnings)` actually includes things that are manually changed into warnings too.

Reply:
>Should be good, I've just confirmed myself that deny(warnings) actually includes things that are manually changed into warnings too.

Yes, and those a lints, as well as dead code ones, however they are not enabled by default, so that could be confusing. The dead code and rust idioms one are the same things when rust blows up on them with error afaics.